File: samples/configdirector-android/java/src/main/java/com/configdirector/sample/java/SampleUser.java
package com.configdirector.sample.java;

import com.configdirector.ConfigDirectorContext;
import java.util.LinkedHashMap;
import java.util.Map;

/**
 * The identities the sample can evaluate configs against. Switching between them calls
 * {@code updateContext}, which reconnects and re-evaluates every config.
 */
enum SampleUser {
  CONFIGURED("Configured", ConfigDirectorContext.builder()
      .id("user-123")
      .name("Sam")
      .trait("plan", "free")
      .build()),

  PRO("Pro plan", ConfigDirectorContext.builder()
      .id("user-456")
      .name("Ada")
      .traits(proTraits())
      .build()),

  ANONYMOUS("Anonymous", ConfigDirectorContext.builder().anonymous(true).build());

  private final String label;
  private final ConfigDirectorContext context;

  SampleUser(String label, ConfigDirectorContext context) {
    this.label = label;
    this.context = context;
  }

  String label() {
    return label;
  }

  ConfigDirectorContext context() {
    return context;
  }

  private static Map<String, Object> proTraits() {
    Map<String, Object> traits = new LinkedHashMap<>();
    traits.put("plan", "pro");
    traits.put("seats", 12);
    traits.put("beta", true);
    return traits;
  }
}

File: samples/configdirector-android/compose/src/main/kotlin/com/configdirector/sample/compose/SampleUser.kt
package com.configdirector.sample.compose

import com.configdirector.ConfigDirectorContext

/**
 * The identities the sample can evaluate configs against. Switching between them calls
 * `updateContext`, which reconnects and re-evaluates every config.
 */
enum class SampleUser(val label: String, val context: ConfigDirectorContext) {
    CONFIGURED("Configured", configuredContext()),

    BETA_TESTER(
        "Beta tester",
        ConfigDirectorContext.build {
            id("beta-tester")
            name("Beta Tester")
            trait("role", "beta")
        },
    ),

    ANONYMOUS("Anonymous", ConfigDirectorContext.build { anonymous(true) }),
}

/** The identity from `local.properties`. With none set, configs are evaluated without a context. */
private fun configuredContext() = ConfigDirectorContext.build {
    id(BuildConfig.USER_ID.ifEmpty { null })
    name(BuildConfig.USER_NAME.ifEmpty { null })
    BuildConfig.USER_ROLE.ifEmpty { null }?.let { role -> trait("role", role) }
}
